core/tee/tee_rpmb_fs.c

```diff
@@ -1499,6 +1524,9 @@ static TEE_Result tee_rpmb_get_write_counter(uint16_t dev_id,
 	if (!counter)
 		return TEE_ERROR_BAD_PARAMETERS;
 
+	if (rpmb_dead)
```

Comment:
How about testing this in `tee_rpmb_invoke()` instead? (note, `tee_rpmb_read()` lacks the check)

Reply:
Testing in `tee_rpmb_invoke()` is too late, then the request is already in non-secure shared memory.
`tee_rpmb_read()` doesn't need this, it's checked before that function is called.

Reply:
OK all good then